pkg/controllers/dynakube/oneagent/daemonset/volumes.go

```diff
@@ -94,8 +116,12 @@ func prepareVolumes(dk *dynakube.DynaKube) []corev1.Volume {
 
 	volumes = append(volumes, getOneAgentSecretVolume(dk))
 
-	if dk.OneAgent().IsReadOnlyOneAgentsMode() {
-		volumes = append(volumes, getCSIStorageVolume(dk))
+	if dk.OneAgent().IsCloudNativeFullstackMode() || dk.OneAgent().IsHostMonitoringMode() {
```

Comment:
```suggestion
	if dk.OneAgent().IsReadOnlyOneAgentsMode() {
```

Reply:
If I am not mistaken this check is exactly covered by `dk.OneAgent().IsReadOnlyOneAgentsMode()`

Reply:
fixed [commit](https://github.com/Dynatrace/dynatrace-operator/pull/4632/commits/c3dbe95e06a37897e68b2f212d55878db995c772)